pkg/kubelet/dockertools/docker_manager_linux.go

```diff
@@ -40,8 +40,13 @@ func getContainerIP(container *dockertypes.ContainerJSON) string {
 func getNetworkingMode() string { return "" }
 
 // Returns true if the container name matches the infrastructure's container name
-func containerProvidesPodIP(name *KubeletContainerName) bool {
-	return name.ContainerName == PodInfraContainerName
+func containerProvidesPodIP(containerName string) bool {
```

Comment:
Yes. Due to limitations on the windows side (mainly the fact that container networking is not supported), we couldn't use the infrastructure container for pods on windows.